open-craft/opencraft
# Managed RabbitMQ

Implements support for a centrally managed RabbitMQ instance when not using the "ephemeral databases" option.

This PR relies on/uses a bunch of other PRs [linked from the ticket](https://tasks.opencraft.com/browse/OC-1719). Also, this code depends on changes made in #145.

**Testing instructions**:

1. Checkout this branch.
1. Add the following to your .env file:
   
   ```
   RABBITMQ_API_URL='https://rabbitmq-dev.opencraft.hosting:15671'
   RABBITMQ_ADMIN_USERNAME='[username]'
   RABBITMQ_ADMIN_PASSWORD='[password]'
   INSTANCE_RABBITMQ_HOST='rabbitmq-dev.opencraft.hosting'
   INSTANCE_RABBITMQ_PORT='5671'
   ```

1. Run `make migrate`.

1. Create a new instance and spawn an appserver. Set the following settings:

   ```
   Configuration source repo url: https://github.com/open-craft/configuration.git
   Configuration version: bdero/opencraft-rabbitmq-vhost-config

   Edx platform repository url: https://github.com/open-craft/edx-platform.git
   Edx platform commit: bdero/opencraft-celery-broker-tls
   ```

   **The following 4 steps can be done before the appserver finishes provisioning.**

1. Go to the [OpenEdXInstance admin panel](http://localhost:5000/admin/instance/openedxinstance/) and check if the vhost and user data was properly generated - this can be verified a couple of minutes after kicking off the appserver spawn task:

   ![screenshot](http://i.imgur.com/XsuYMZ5.png)

1. Navigate to https://rabbitmq-dev.opencraft.hosting:15671 in your browser and login to the administration account. The administration account credentials can be found by decrypting the [rabbitmq-dev host vars](https://github.com/open-craft/deployment-secrets-services/pull/13/files#diff-0b5b5280556d5a19a034420a0c90b5feR1).

1. Select the "Admin" tab and verify that the two new users were added to the new vhost.

1. Now, go grab some coffee/tea/beer. It's time to wait for the appserver to finish provisioning.

1. Once the appserver finishes provisioning, login to studio on the new Open edX instance and create a rerun of the default course.

1. Return to the RabbitMQ admin panel and verify that there was activity on the instance's new vhost.

**Reviewers**
